deepcell/image_generators/semantic.py

```diff
@@ -85,7 +90,16 @@ def __init__(self,
                  save_to_dir=None,
                  save_prefix='',
                  save_format='png'):
+        # Load data
+        if 'X' not in train_dict.keys():
+            raise ValueError('No training data found in train_dict')
+
+        if 'y' not in train_dict.keys():
```

Comment:
Calling `.keys()` actually creates a new list. It's preferred to just check if they key is in the dictionary:

```python
if 'X' not in train_dict:
```

Reply:
```suggestion
        if 'X' not in train_dict:
            raise ValueError('No training data found in train_dict')

        if 'y' not in train_dict:
```